OpenSim/Common/Component.cpp

```diff
@@ -192,6 +192,39 @@ void Component::finalizeFromProperties()
 
     markPropertiesAsSubcomponents();
     componentsFinalizeFromProperties();
+
+    auto subs = getImmediateSubcomponents();
+    std::set<std::string> names{};
+    std::string name{};
+
+    int count = 0;
+    for (auto& sub : subs) {
+        name = sub->getName();
+        const auto& type = sub->getConcreteClassName();
+        auto it = names.find(type+name);
+
+        if (it == names.cend())
+            names.insert(type + name);
+        else {
+            // In the future this should become an Exception 
+            //OPENSIM_THROW(SubcomponentsWithDuplicateName, getName(), name);
+            
+            //for now rename the duplicately named Component
+            Component* mutableSub = const_cast<Component *>(sub.get());
+            mutableSub->setName(name + "_" + std::to_string(count));
+
+            // Warn of the problem
+            std::string msg = type + " '" + getName() + "' has subcomponents " +
+                "with duplicate name '" + name + "', and was renamed as '" +
+                sub->getName() + "'.";
+            std::cout << msg << std::endl;
+
+            names.insert(type + name);
```

Comment:
Also, if `count` is initialized to `0` before this `while(1)`, non-unique names like "p1, p2, p2, p2" will be renamed "p1, p2, p2_0, p2_1" rather than "p1, p2, p2_17, p2_18" (or however many subcomponents have already been visited).